suspendmanager.lua

```diff
@@ -531,3 +557,42 @@ end
 if not dfhack_flags.module then
     main({...})
 end
+
+-- Overlay Widget
+JobOverlay = defclass(JobOverlay, overlay.OverlayWidget)
+JobOverlay.ATTRS{
+    default_pos={x=-41,y=14},
+    default_enabled=true,
+    viewscreens='dwarfmode/ViewSheets/BUILDING',
+    frame={w=30, h=5},
+    frame_style=gui.MEDIUM_FRAME,
+    frame_background=gui.CLEAR_PEN,
+}
+
+function JobOverlay:init()
+    self:addviews{
+        widgets.Label{
+            frame={t=0, l=0},
+            text='Staying suspended:'
+        },
+        widgets.Label{
+            frame={t=2, l=0},
+            text={'', {text=self:callback('get_reason_string')}}
+        }
+    }
+end
+
+function JobOverlay:get_reason_string()
+    return Instance:suspensionDescription(dfhack.gui.getSelectedJob())
+end
+
+function JobOverlay:render(dc)
+    if not isEnabled() or not self:get_reason_string() then
```

Comment:
the suspension reason message should probably still be displayed even when `suspendmanager` is not enabled -- the building could have been suspended by `suspendmanager` and then the player temporarily disabled `suspendmanager`.

if `suspendmanager` is disabled, it might be good to put a HotkeyLabel here that enables `suspendmanager` (as long as the button isn't too intrusive or pushy)

Reply:
Oh, I like the idea of the button here. It goes both way too, to allow disabling it from here. If a player is annoyed by `suspendmanager` suspending a job they get both a reason why and a way to bypass the behavior, or if they actually want to manually suspend some jobs, this is the first place they will search for too.

As far as always displaying the reason, I don't think it a good idea, when suspendmanager is disabled the reason will become out of date, maybe the player even unsuspended and resuspended the job. Essentially when suspendmanager is disabled, we don't have a clue of the reason of suspension.